librb/include/rb_memory.h

```diff
@@ -37,6 +37,7 @@
 
 void rb_outofmemory(void) __attribute__((noreturn));
 
+__attribute__((malloc, alloc_size(1), returns_nonnull))
```

Comment:
In work. But I think there are other attribute crimes abound. Also this is only ever (and should only ever) be used here, so for a future compiler it would be cleaner to just #ifdef the attribute rather than #ifdef the macro that is used in one place?